tests/migrations/test_operations.py

```diff
@@ -2092,96 +2100,109 @@ def test_remove_partial_unique_constraint(self):
         self.assertEqual(
             operation.describe(),
-            'Create constraint deferrable_pink_constraint '
-            'on model Pony'
+            'Create constraint deferred_pink_constraint_add on model Pony',
         )
-        # Test the state alteration
+        # Add constraint.
         new_state = project_state.clone()
         operation.state_forwards(app_label, new_state)
         self.assertEqual(len(new_state.models[app_label, 'pony'].options['constraints']), 1)
         Pony = new_state.apps.get_model(app_label, 'Pony')
         self.assertEqual(len(Pony._meta.constraints), 1)
-        # Test the database alteration
         with connection.schema_editor() as editor:
             operation.database_forwards(app_label, editor, project_state, new_state)
-        # Test constraint works
         Pony.objects.create(pink=1, weight=4.0)
-        with atomic():
+        if connection.features.supports_deferrable_unique_constraints:
+            # Unique constraint is deferred.
+            with transaction.atomic():
+                obj = Pony.objects.create(pink=1, weight=4.0)
+                obj.pink = 2
+                obj.save()
+            # Constraint behavior can be changed with SET CONSTRAINTS.
+            with self.assertRaises(IntegrityError):
+                with transaction.atomic(), connection.cursor() as cursor:
+                    quoted_name = connection.ops.quote_name(deferred_unique_constraint.name)
+                    cursor.execute('SET CONSTRAINTS %s IMMEDIATE' % quoted_name)
+                    obj = Pony.objects.create(pink=1, weight=4.0)
+                    obj.pink = 3
+                    obj.save()
+        else:
             Pony.objects.create(pink=1, weight=4.0)
-            if connection.features.supports_deferrable_unique_constraints:
-                with self.assertRaises(IntegrityError):
-                    cursor = connection.cursor()
-                    cursor.execute(
-                        'SET CONSTRAINTS {} IMMEDIATE'.format(
-                            connection.ops.quote_name(deferrable_unique_constraint.name)))
-        # Test reversal
+        # Reversal.
         with connection.schema_editor() as editor:
             operation.database_backwards(app_label, editor, new_state, project_state)
-        # Test constraint doesn't work
+        # Constraint doesn't work.
         Pony.objects.create(pink=1, weight=4.0)
-        # Test deconstruction
+        # Deconstruction.
         definition = operation.deconstruct()
         self.assertEqual(definition[0], 'AddConstraint')
         self.assertEqual(definition[1], [])
-        self.assertEqual(definition[2], {'model_name': 'Pony', 'constraint': deferrable_unique_constraint})
+        self.assertEqual(
+            definition[2],
+            {'model_name': 'Pony', 'constraint': deferred_unique_constraint},
+        )
+        definition = operation.deconstruct()
```

Comment:
```suggestion
```

Reply:
Good catch, I will remove it before final squash.